### PR TITLE
Add the ability to specify additional import directories

### PR DESCRIPTION
**Pull request by jhaber**
This adds an option to the plugin to allow users to specify additional import directories. The protos in these directories won't be compiled, but they're available for import in the protos that do get compiled. 

For some background on our use-case, we use this plugin to compile protos at build time and we package the `.proto` sources in addition to the compiled proto classes in the JAR. Having one proto import another works fine when they're in the same Maven module, but it is difficult to import across module boundaries. The way I would like this to work is to copy all of the dependency `.proto` files into subfolders based on the coordinates. These would be available to import, but not compiled or packaged with the jar. So if groupA:artifactA provides example.proto, the idea is you could easily import it in groupB:artifactB by doing `import groupA/artifactA/example.proto`. I think being able to specify additional import directories is the only thing needed to support this use-case (we can handle the rest in a separate plugin).

Thanks in advance, and let me know if you have any feedback. The one question I kept going back-and-forth on is whether `importDirectories` should automatically get included as `testImportDirectories` (similar to how compile/runtime deps are automatically on the test classpath)

### CODE REVIEW

**Comment by comoyouser** [comment]
Can one of the admins verify this patch?

**Comment by jhaber** [comment]
I spent some time trying to get the dependency imports working like I described using our fork of the plugin and protoc really didn't like it (I think the problem was the directory structure changing between the original compilation and when the protobufs are imported as a dependency). I did get it working using a slightly different strategy though. We copy the dependency protobufs into a directory like before, and then users can explicitly add the dependencies they want to import from using `importDirectories`. So using the same example as before (groupA:artifactA provides example.proto), you'd add a pom snippet like this to groupB:artifactB

```xml
<pluginManagement>
  <plugins>
    <plugin>
      <groupId>com.comoyo.maven.plugins</groupId>
      <artifactId>protoc-bundled-plugin</artifactId>
      <configuration>
        <importDirectories>
          <importDirectory>${dependency-protobufs.folder}/groupA/artifactA</importDirectory>
        </importDirectories>
      </configuration>
    </plugin>
  </plugins>
</pluginManagement>
```

And then you can `import example.proto`

tl;dr - this is a long way of saying this PR is still useful. And I think this also settles the question of whether `importDirectories` should automatically get included as `testImportDirectories` (they shouldn't in my opinion)

**Comment by jhaber** [comment]
I just ran into an issue which is that I can't import built-in proto definitions like google.protobuf.Any, google.protobuf.Duration, google.protobuf.FieldMask, etc. Incidentally, this PR helps to work around the problem. You can use it in conjunction with the [copy-protos-maven-plugin](https://github.com/HubSpot/copy-protos-maven-plugin). So to get these imports working would look like:

```xml
<build>
    <pluginManagement>
        <plugins>
            <plugin>
                <groupId>com.hubspot.maven.plugins</groupId>
                <artifactId>copy-protos-maven-plugin</artifactId>
                <version>0.1</version>
            </plugin>
            <plugin>
                <groupId>com.comoyo.maven.plugins</groupId>
                <artifactId>protoc-bundled-plugin</artifactId>
                <version>1.4-hubspot-SNAPSHOT</version>
                <configuration>
                    <protobufVersion>${dep.protobuf-java.version}</protobufVersion>
                    <importDirectories>
                        <importDirectory>${project.build.directory}/generated-resources/dependency-protobufs/com/google/protobuf/protobuf-java</importDirectory>
                    </importDirectories>
                </configuration>
            </plugin>
        </plugins>
    </pluginManagement>

    <plugins>
        <plugin>
            <groupId>com.hubspot.maven.plugins</groupId>
            <artifactId>copy-protos-maven-plugin</artifactId>
            <executions>
                <execution>
                    <id>copy-dependency-protos</id>
                    <goals>
                        <goal>copy-protos</goal>
                    </goals>
                </execution>
            </executions>
        </plugin>
        <plugin>
            <groupId>com.comoyo.maven.plugins</groupId>
            <artifactId>protoc-bundled-plugin</artifactId>
            <executions>
                <execution>
                    <goals>
                        <goal>run</goal>
                    </goals>
                </execution>
            </executions>
        </plugin>
    </plugins>
</build>
```

Ideally this wouldn't be so verbose and wouldn't involve multiple plugins, but at least it works and solves the problem. Another enhancement could be to integrate this functionality into this plugin directly so that you could do something like:

 ```xml
<build>
    <pluginManagement>
        <plugins>
            <plugin>
                <groupId>com.comoyo.maven.plugins</groupId>
                <artifactId>protoc-bundled-plugin</artifactId>
                <version>1.4-hubspot-SNAPSHOT</version>
                <configuration>
                    <protobufVersion>${dep.protobuf-java.version}</protobufVersion>
                    <importDependencies>
                        <importDependency>com.google.protobuf:protobuf-java</importDependency>
                    </importDependencies>
                </configuration>
            </plugin>
        </plugins>
    </pluginManagement>

    <plugins>
        <plugin>
            <groupId>com.comoyo.maven.plugins</groupId>
            <artifactId>protoc-bundled-plugin</artifactId>
            <executions>
                <execution>
                    <goals>
                        <goal>run</goal>
                    </goals>
                </execution>
            </executions>
        </plugin>
    </plugins>
</build>
```

**Comment by argggh** [comment]
First of all, apologies for letting this linger for so long. Since no-one at Telenor Digital is actively working on this project today, there was a hiatus while we worked to resolve the governance situation. Having moved the code base to an independent github organization unblocks this, and I'll be looking into this PR (again) now. If there are any updates since January I'd be happy to be notified.
